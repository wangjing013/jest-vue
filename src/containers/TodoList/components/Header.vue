<template>
  <div class="header">
    <div class="content">
      TodoList
      <input
        class="input"
        data-test="input"
        v-model="inputValue"
        @keyup.enter="handlerEnter"
        placeholder="请输入代办事项"
      />
    </div>
  </div>
</template>

<script>
export default {
  name: 'Header',
  data() {
    return {
      inputValue: ''
    }
  },
  methods: {
    handlerEnter() {
      if (this.inputValue) {
        this.$emit('add', this.inputValue)
        this.inputValue = ''
      }
    }
  }
}
</script>

<style scoped lang="stylus">
.header {
  background-color #333
  height 60px
  color #fff
}
.content {
  width 600px
  margin: 0 auto
  line-height 60px
}
.input {
  outline none
  border none
  height 32px;
  line-height 32px;
  width 400px
  padding 6px 12px;
  float right
  margin-top 14px
}
</style>
