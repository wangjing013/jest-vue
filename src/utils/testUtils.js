export const findTestWarpper = (warpper, selector) => {
  return warpper.find(`[data-test="${selector}"]`)
}
