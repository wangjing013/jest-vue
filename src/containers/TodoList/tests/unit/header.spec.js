import { shallowMount } from '@vue/test-utils'
import Header from '../../components/Header.vue'
import { findTestWarpper } from '@/utils/testUtils'

let warpper = null
beforeAll(() => {
  warpper = shallowMount(Header)
})

it('header 样式添加快照', () => {
  expect(warpper).toMatchSnapshot('')
})

it('header 包含input框', () => {
  const input = findTestWarpper(warpper, 'input')
  expect(input.exists()).toBe(true)
})

it('input 框初始值应该为空', () => {
  expect(warpper.vm.$data.inputValue).toBe('')
})

it('当 input 值发生改变时, data 中值也发生改变', async() => {
  const input = findTestWarpper(warpper, 'input')
  const value = 'some value'
  await input.setValue(value)
  expect(warpper.vm.$data.inputValue).toBe(value)
})

it('当 input 内容为空时,回车时不对外发送事件', async() => {
  const input = findTestWarpper(warpper, 'input')
  await input.setValue('')
  input.trigger('keyup.enter')
  expect(warpper.emitted().add).toBeFalsy()
})

it('当 input 内容不为空时,回车对外外发送事件', async() => {
  const input = findTestWarpper(warpper, 'input')
  await input.setValue('some value')
  input.trigger('keyup.enter')
  expect(warpper.emitted().add).toBeTruthy()
})

it('当 input 内容不为空时,回车时清空内容', async() => {
  const input = findTestWarpper(warpper, 'input')
  await input.setValue('some value')
  await input.trigger('keyup.enter')
  expect(warpper.vm.$data.inputValue).toBe('')
})
