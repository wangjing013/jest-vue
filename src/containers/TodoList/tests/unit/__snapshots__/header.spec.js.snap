// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 组件 样式发生变化时,有提示 1`] = `
<div class="header">
  <div class="content">
    TodoList
    <input data-test="input" placeholder="请输入代办事项" class="input">
  </div>
</div>
`;
